pandas/core/series.py

```diff
@@ -2809,6 +2810,22 @@ def _try_cast(arr, take_fast_path):
             subarr = maybe_cast_to_datetime(arr, dtype)
             if not is_extension_type(subarr):
                 subarr = np.array(subarr, dtype=dtype, copy=copy)
+
```

Comment:
I am going to move this entire section (e.g. ``_try_cast``) to ``pandas/types/cast``. but I will do that after